tests/foreman/cli/test_remoteexecution.py

```diff
@@ -134,6 +134,46 @@ def test_positive_run_default_job_template(
         assert 'Exit' in out
         assert 'Internal Server Error' not in out
 
+    @pytest.mark.tier3
+    @pytest.mark.rhel_ver_list([8])
+    def test_negative_run_default_job_template(
+        self, module_org, rex_contenthost, module_target_sat
+    ):
+        """Confirm that job status is correct if remote action fails
+
+        :id: b6229222-26cf-46bd-bbf6-46814cafd085
+
+        :expectedresults: Verify the job exits as expected
+
+        :verifies: SAT-28435
```

Comment:
I hesitate to mark this as regression test for that bug. This is quite a general test that jobs report failure correctly. Actually, I see multiple tests that test this, although the failure is caused by something else (`test_negative_time_to_pickup`, `test_positive_ansible_job_with_nonexisting_module`, `test_positive_run_job_in_chosen_directory`). So I need to ask,
1) Why didn't these tests catch the regression?
2) Isn't it better to fix these tests? Just a thought, I personally think having this general and simple test is a good thing.
3) Shouldn't you write yet another test that reproduces the regression the way it's described in the bug?

Reply:
The thing is that the issue is not occurring on ansible style rex, nor pull provider rex, that's why the existing negative tests didn't catch it. The bug describes a scenario that uncovered it, but in fact any job failure on client reproduces it, the fix is general as well https://github.com/theforeman/smart_proxy_remote_execution_ssh/pull/119.